src/bun.js/module_loader.zig

```diff
@@ -2314,19 +2315,48 @@ pub const ModuleLoader = struct {
             }
         }
 
-        const synchronous_loader = lr.loader orelse loader: {
+        const synchronous_loader: options.Loader = lr.loader orelse loader: {
             if (jsc_vm.has_loaded or jsc_vm.is_in_preload) {
                 // Extensionless files in this context are treated as the JS loader
                 if (lr.path.name.ext.len == 0) {
-                    break :loader options.Loader.tsx;
+                    break :loader .tsx;
                 }
 
                 // Unknown extensions are to be treated as file loader
-                break :loader options.Loader.file;
+                if (is_commonjs_require) {
+                    if (jsc_vm.commonjs_custom_extensions.entries.len > 0 and
+                        jsc_vm.has_mutated_built_in_extensions == 0)
+                    {
+                        @branchHint(.unlikely);
+                        if (node_module_module.findLongestRegisteredExtension(jsc_vm, lr.path.text)) |entry| {
```

Comment:
de-duplicate this + matching logic in `Bun__transpileFile`

Reply:
there is no logic worth deduping